### PR TITLE
Random Event Tweaks: Preparing for the Overhauls

### PR DESCRIPTION
**Pull request by JohnWildkins**
part 1 of many changes to make events more interesting

- Added:
  - Rescue Pod (moderate event); replaces the Rescue Pod ghost spawner, otherwise operates the same.
  - Burglar Pod (moderate event); mutually exclusive subtype of Rescue Pod, occurs only during non-extended rounds. Sends a single burglar in an escape pod toward the Aurora.
  - Visitor (minor event); replaces the Visitor ghost spawner; otherwise operates the same.

- Event chance increased:
  - Radiation Storm
  - Electrical Storm
  - Communication Blackout

- Event chance reduced:
  - Moderate Vermin/Spider Infestation
  - Rogue Drones
  - Carp School (Moderate)
  - Rampant Brand Intelligence (pending rework on acceptance of #10311; now scales off of engineers rather than janitors, and cannot appear without an engineer slotted)
  - Appendicitis (now scales off of Surgeons rather than all medical staff)
 
- Events removed:
  - Grid Check (laggy and essentially identical to electrical storm)

- Other tweaks:
  - Events that fail to spawn due to certain conditions (rescue pod, blob, etc.) now will attempt to spawn a different event of the same severity tier.

essentially the most common events all happened to be drawing off of ASSIGNMENT_SECURITY (which, as you might expect in SecStation 13, meant that basically every round had one or more Sec-focused event at once). Reduced the scaling of that to be more in line with other events given our normal security population, and some other tweaks.

look to the future for:
- better existing events
- new events, especially in the moderate and major tier
- frameworks to make event selection more varied, and more in-depth event chains

### CODE REVIEW

**Comment by JohnWildkins** [comment]
!review

File: code/modules/events/rescue_pod.dm [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+/datum/event/rescue_pod/burglar
+    no_fake = 1
+    spawner_name = "burglarpod"
+
+/datum/event/rescue_pod/burglar/announce()
+    if(prob(33)) // Make a silent drop more likely since they're spooky burglar
+        ..()
+
+/datum/event/rescue_pod/burglar/setup()
+    . = ..()
```
Comment (mikomyazaki) [inline code/modules/events/rescue_pod.dm:30]: This override does not need to exist, it's empty.

File: code/modules/events/visitor.dm [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/datum/event/visitor
+    no_fake = 1
+    var/datum/ghostspawner/human/visitor/spawner
+
+/datum/event/rescue_pod/setup()
```
Comment (mikomyazaki) [inline code/modules/events/visitor.dm:5]: Should this be /datum/event/visitor/setup()?

Same for start() below.

**Review by mikomyazaki — CHANGES_REQUESTED** [review]

File: code/modules/events/visitor.dm (hunk 2 above)
Comment (JohnWildkins) [reply code/modules/events/visitor.dm:5]: yes, me tired, good catch

**Review by mikomyazaki — APPROVED** [review]

File: code/modules/events/rescue_pod.dm [hunk 3]
```diff
@@ -0,0 +1,31 @@
+/datum/event/rescue_pod
+    no_fake = 1
+    var/spawner_name = "rescuepodsurv"
+    var/datum/ghostspawner/human/rescuepodsurv/spawner
+
+/datum/event/rescue_pod/announce()
+    if(prob(66))
+        command_announcement.Announce("The NDV Icarus reports an escape pod inbound to the [station_name()]. Investigate and attend to the situation in accordance with NanoTrasen corporate regulations.", new_title="NDV Icarus")
```
Comment (NonQueueingMatt) [inline code/modules/events/rescue_pod.dm:8]: if possible this should have a soundclip - also maybe the burglar one shouldn't happen on extended

File: code/modules/events/event_container.dm [hunk 4]
```diff
@@ -144,29 +145,30 @@ var/global/list/severity_to_string = list(EVENT_LEVEL_MUNDANE = "Mundane", EVENT
...
 
 /datum/event_container/moderate
 	severity = EVENT_LEVEL_MODERATE
 	available_events = list(
 		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Nothing",							/datum/event/nothing,						200),
-		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Carp School",						/datum/event/carp_migration,				50, 	list(ASSIGNMENT_SECURITY = 25)),
-		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Communication Blackout",			/datum/event/communications_blackout,		60),
-		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Electrical Storm",					/datum/event/electrical_storm, 				50,		list(ASSIGNMENT_ENGINEER = 5, ASSIGNMENT_JANITOR = 20)),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Carp School",						/datum/event/carp_migration,				25, 	list(ASSIGNMENT_SECURITY = 15)),
+		new /datum/event_meta(EVENT_LEVEL_MODERATE, "Communication Blackout",			/datum/event/communications_blackout,		100),
```
Comment (NonQueueingMatt) [inline code/modules/events/event_container.dm:157]: Not sure I agree with increasing the chances of these. They're way more disruptive than carp schools, even. Why'd you opt for this?

File: code/modules/events/rescue_pod.dm (hunk 3 above)
Comment (JohnWildkins) [reply code/modules/events/rescue_pod.dm:8]: the burglar one already doesn't happen on extended

File: code/modules/events/event_container.dm (hunk 4 above)
Comment (JohnWildkins) [reply code/modules/events/event_container.dm:157]: It's only about a 2% per event roll increase, and Electrical Storm wasn't really increased so much as it took about half of Grid Check's chance to happen. If they're overtuned it's not much of an issue to knock them back down, but as of now they're still about half as likely to happen as Carp School.

**Review by Geevies — APPROVED** [review]
random chimp event when

**Pull request merged**